packages/state-transition/src/cache/epochCache.ts

```diff
@@ -246,13 +261,14 @@ export class EpochCache {
    */
   static createFromState(
     state: BeaconStateAllForks,
-    {config, pubkey2index, index2pubkey}: EpochCacheImmutableData,
+    {config, finalizedPubkey2index, finalizedIndex2pubkey}: EpochCacheImmutableData,
+    unfinalizedPubkey2index: UnfinalizedPubkeyIndexMap,
```

Comment:
Is `createFromState` ever called with a non-finalized state? If it is created with a non-finalized state you expect the caller to pass the correct `unfinalizedPubkey2index`, which is not check here to be correct. This looks like a complicated pattern. Don't support it if it's not necessary.

Reply:
> Is `createFromState` ever called with a non-finalized state? If it is created with a non-finalized state you expect the caller to pass the correct `unfinalizedPubkey2index`, which is not check here to be correct. This looks like a complicated pattern. Don't support it if it's not necessary.

No, currently all callers are calling with finalized state. I was debating whether to add `unfinalizedPubkey2index` into the param because technically `createFromState()` can take in non-finalized state. I will remove this change for now, and will re-introduce it in the future if necessary